### PR TITLE
Fixing the last final critical issue

### PR DESCRIPTION
**Pull request by rafaelweingartner**
I hope we will not have anyone else for a long time.
@GabrielBrascher, @pedro-martins would you mind checking this one too?

### CODE REVIEW

**Comment by GabrielBrascher** [comment]
I haven't found any issue to point. Code LGTM.

**Comment by rafaelweingartner** [comment]
@GabrielBrascher is it better now the diff from "AutonomiccsSystemVmsGuru" class?

**Comment by GabrielBrascher** [comment]
@rafaelweingartner nice, could you please squash those commits? Thanks.

File: autonomic-algorithms-commons/src/main/java/br/com/autonomiccs/autonomic/algorithms/commons/resources/HostResources.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 public class HostResources implements Cloneable {
 
-    private List<VmResources> vmsResources = new ArrayList<VmResources>();
+    private List<VmResources> vmsResources = new ArrayList<>();
     private float cpuOverprovisioning, memoryOverprovisioning;
```
Comment (pedro-martins) [inline autonomic-algorithms-commons/src/main/java/br/com/autonomiccs/autonomic/algorithms/commons/resources/HostResources.java:32]: @rafaelweingartner I don't understand why you remove the type from the constructor, both are correct I know but in older versions of java (6 and older) , the type in a list constructor is necessary, at the last this code won't compile in java 6 or older versions.

Comment (rafaelweingartner) [inline autonomic-algorithms-commons/src/main/java/br/com/autonomiccs/autonomic/algorithms/commons/resources/HostResources.java:32]: You are right. But, the point is that, we already use java 7, and we have to move to java 1.8. Oracle ceased public availability of security fixes and upgrades for Java 7 as of April 2015. We will not move backward.

**Comment by rafaelweingartner** [comment]
@GabrielBrascher, @pedro-martins done.
If you think it is ok, please go ahead and merge.

**Comment by pedro-martins** [comment]
I think this is ok now. Thanks for your explanation =P

**Pull request merged**